### PR TITLE
Better json environment runner

### PR DESCRIPTION
**Pull request by albertchen-sifive**
use the new environment-example-sifive runner from this PR https://github.com/sifive/environment-example-sifive/pull/1

### CODE REVIEW

**Comment by richardxia** [comment]
Sorry, I totally forgot about this PR. Let's do this before the next release (after the 0.4.0 that I'll cut soon).

**Comment by richardxia** [comment]
Sorry for taking such a long time to get back to this. I've rebased this on top of master, and I made a few more changes:

- https://github.com/sifive/environment-blockci-sifive/pull/5/commits/3608a3843c421d564c9d8bfec28049cc607e7180 changes the specific resource versions to match the actual versions installed in the Docker image
- https://github.com/sifive/environment-blockci-sifive/pull/5/commits/d1d4bba1004572f00dcaaaa0ae3e5412795c1e43 bumps the wit manifest pointer to environment-example-sifive to the most recent tagged version ([v0.2.0](https://github.com/sifive/environment-example-sifive/releases/tag/v0.2.0)). This also pulls in a version that supports wake 0.18.
- https://github.com/sifive/environment-blockci-sifive/pull/5/commits/2a54c47388801b0ef98f51900d30de0918fe4a41 adds a CHANGELOG entry that says that this environment package now only matches on specific versions of tools

Note that this will _totally break_ compatibility with api-generator-sifive and its dependent repositories, since the resource versions listed in environment.json don't actually match what the Wake rules currently request: api-firrtl-sifive wants protobuf 3.5.1, api-generator-sifive wants Verilator 4.028, and freedom-metal wants riscv-tools 2019.05. I actually have to _downgrade_ the versions of several of these tools in the Dockerfile in order to match the requested versions. I'm planning to do this in a separate PR just to keep this scoped to migrating to a JSON configuration value.

I've been testing this against block-pio-sifive, which is how I discovered the version discrepancies. Before I tag this repo with 0.5.0, I want to get the version bumps (and unbumps) of the three aforementioned tools, just so that we get into a consistent state. I'm thinking of cutting an 0.6.0 with wake 0.18, which leaves 0.5.0 as an intermediate version in case if there are issues with the tool version changes without also having to deal with the Wake version changes.

File: wake/environment.wake [hunk 1]
```diff
@@ -1,78 +1 @@
-#################################################################################
-## Logic for handling custom environment
-##
-## Do not edit below this line
-#################################################################################
```
Comment (mmjconolly) [inline wake/environment.wake:5]: Is this similar to what was in environment-example-sifive?

Comment (richardxia) [reply wake/environment.wake:5]: Yeah, this essentially got factored out _into_ environment-example-sifive in https://github.com/sifive/environment-example-sifive/pull/1 (both this and that PR were submitted in the middle of last November).

**Review by mmjconolly — APPROVED** [review]

**Pull request merged**